### PR TITLE
Tmp/mwang/tunneler ci2

### PR DESCRIPTION
**Pull request by miawangTT**
### Problem description
Add tunneler tests to micro benchmark test so that they can be run on BM machines.

### Checklist
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by TT-billteng — APPROVED** [review]

**Review by blozano-tt — APPROVED** [review]

**Comment by tt-rkim** [comment]
Are these deterministically passing now?

If you would like to do extra testing, I recommend using the custom test dispatch workflow to run your test in a loop on our VM infrastructure

**Comment by miawangTT** [comment]
Thanks for offering. We move this test from post-commit to micro-benchmark pipeline to make it run on BM machines instead of VMs.

**Comment by miawangTT** [comment]
Thanks for letting me know! The codes have been modified to get rid of the prior failure (which is related to how fast a node can be set up on the machine). And it has been tested with for loops on both local and CI machines. The tests only take a few minutes to run.

**Review by tt-rkim — APPROVED** [review]

**Comment by tt-rkim** [comment]
How was running it? Did it look ok / stable?

**Comment by miawangTT** [comment]
Yes, it is stable https://github.com/tenstorrent/tt-metal/actions/runs/12044418841

**Review by ubcheema — APPROVED** [review]

**Comment by tt-rkim** [comment]
It's been a while

Everything OK?

File: .github/workflows/metal-run-microbenchmarks.yaml [hunk 1]
```diff
@@ -42,6 +42,8 @@ jobs:
           PIPELINE_TYPE="microbenchmarks"
           if [ "${{ matrix.runner-info.ccl }}" == "true" ]; then
             PIPELINE_TYPE="ccl_microbenchmarks"
+          else
+            TT_METAL_SLOW_DISPATCH_MODE=1 ./tests/scripts/run_tunneler_tests.sh --machine-type ${{ matrix.runner-info.runs-on[0] }}
```
Comment (tt-rkim) [inline .github/workflows/metal-run-microbenchmarks.yaml:46]: Ooohh that's smart but a dangerous assumption...  we should be careful to run all configs of this to be sure we don't lose track of `[0] == "N300"` in this case

**Review by tt-rkim — APPROVED** [review]

**Pull request merged**